django/db/models/base.py

```diff
@@ -1665,6 +1666,25 @@ def _check_long_column_names(cls):
 
         return errors
 
+    @classmethod
+    def _check_indexes(cls):
+        """
+        Check fields in "indexes" option are valid.
+        """
+        errors = []
+        field_names = [field.name for field in cls._meta.fields]
+        index_field_names = [field_name for index in cls._meta.indexes for field_name in index.fields]
```

Comment:
so i think it would be better if there were actual nested for-loops:

```py
for index in cls._meta.indexes:
    for name, order in index.fields_orders:  # avoid the '-' logic in the check, it's already done in Index.__init__
```

and if we used `obj=index` then the message should be clearer as it shows which index is referred to